source/val/instruction.cpp

```diff
@@ -19,34 +19,27 @@
 namespace spvtools {
 namespace val {
 
-#define OPERATOR(OP)                                                 \
-  bool operator OP(const Instruction& lhs, const Instruction& rhs) { \
-    return lhs.id() OP rhs.id();                                     \
-  }                                                                  \
-  bool operator OP(const Instruction& lhs, uint32_t rhs) {           \
-    return lhs.id() OP rhs;                                          \
-  }
-
-OPERATOR(<)
-OPERATOR(==)
-#undef OPERATOR
-
-Instruction::Instruction(const spv_parsed_instruction_t* inst,
-                         Function* defining_function,
-                         BasicBlock* defining_block)
+Instruction::Instruction(const spv_parsed_instruction_t* inst)
     : words_(inst->words, inst->words + inst->num_words),
       operands_(inst->operands, inst->operands + inst->num_operands),
       inst_({words_.data(), inst->num_words, inst->opcode, inst->ext_inst_type,
              inst->type_id, inst->result_id, operands_.data(),
-             inst->num_operands}),
-      line_num_(0),
-      function_(defining_function),
-      block_(defining_block),
-      uses_() {}
+             inst->num_operands}) {}
 
 void Instruction::RegisterUse(const Instruction* inst, uint32_t index) {
   uses_.push_back(std::make_pair(inst, index));
 }
 
+bool operator<(const Instruction& lhs, const Instruction& rhs) {
+  return lhs.id() < rhs.id();
```

Comment:
I see where this is a port from, but this doesn't seem correct. Do you know where this is used? Only using result id will cause all stores to appear equal for example.

Reply:
I have no idea, I can do a followup CL to attempt to remove them and see what happens

Reply:
SGTM